tracer/src/Datadog.Trace/Agent/StatsAggregator.cs

```diff
@@ -16,10 +17,10 @@ internal class StatsAggregator : IStatsAggregator
     {
         private const int BufferCount = 2;
 
-        private const int BucketDurationSeconds = 10;
-
         private static readonly IDatadogLogger Log = DatadogLogging.GetLoggerFor<StatsAggregator>();
 
+        private readonly HashSet<StatsAggregationKey> _keys;
```

Comment:
I have a hard time estimating the cardinality of a key. I assume it's driven mainly by resource right? We shouldn't have that many of them hopefully. I assume we don't but would we have this data somewhere?

Reply:
Yeah, this naïve implementation could start to build up without a way to limit the set. I think you're right that it's mainly driven by the cardinality of resource names. Here's how a key is composed, which is a unique combination of the following fields:
 - Resource: Most likely to be high-cardinality
 - Service: Low cardinality
 - OperationName: Low cardinality?
 - Type: Low cardinality?
 - HttpStatusCode: Low cardinality. It's only expected on type `http` (0 otherwise) and I expect the range of status codes to be limited
 - IsSyntheticsRequest: Low cardinality. `true/false`

Reply:
Yep so resource is the one with the high cardinality (plus I suspect that will be the longest string as well). So the main factor for the size of this set will be resources, no? 

Maybe there's a metric the stats team emit to follow the number of resources, that would give us an idea. We could ask on #apm-stats just in case. 
And to be clear, without more data, I'm not challenging the implementation (best is the enemy of good)

Reply:
Just as a reminder, there's an ongoing issue with high cardinality of resource names for outgoing http requests that contain short identifier segments. There could be _thousands_ of them 😬